api/v0/api.go

```diff
@@ -0,0 +1,558 @@
+package v0
+
+import (
+	"context"
+	datatransfer "github.com/filecoin-project/go-data-transfer"
+	"github.com/filecoin-project/go-state-types/network"
+	"github.com/filecoin-project/lotus/api" // todo: alias types from here
```

Comment:
It may also make more sense to move types here, then alias from newer versions, kind of like we do in actors - then the newer versions only need to declare new/changing types, and it's harder to break older apis by changing things in new apis

The drawback is that we can't easily disable building of those wrapper layers, potentially making binaries a bit bigger than ideal, but I don't think that's a big issue